### PR TITLE
build: initial dashboard project

### PR DESCRIPTION
**Pull request by DevVersion**
* Creates the initial dashboard project using the Angular CLI (minimalistic and only necessary parts)
* Updates the deployment script to upload the dashboard inside of Travis

### CODE REVIEW

File: scripts/deploy/deploy-dashboard.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
```
Comment (jelbourn) [inline scripts/deploy/deploy-dashboard.sh]: Add a top-level comment mentioning what this script does ("Deploys dashboard app to Firebase, both hosting and functions").

File: scripts/deploy/deploy-dashboard.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+cd $(dirname ${0})/../..
+
+# Paths to the dashboard and functions directories.
+dashboardFolder=tools/dashboard
+
+# Go to the dashboard folder because otherwise the Firebase CLI tries to deploy the wrong project.
+cd ${dashboardFolder}
+
+# Install node_modules for the dashboard and afterwards build the dashboard app in production.
+(npm install; npm run build-production) &
```
Comment (jelbourn) [inline scripts/deploy/deploy-dashboard.sh]: Can we make this just `ng build --aot --prod` here? Everyone knows that that does but the `npm run` adds a level of indirection

File: tools/dashboard/src/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "lib": ["es2016", "dom"],
+    "sourceMap": true,
+    "declaration": false,
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "target": "es5",
+    "module": "es2015",
+    "typeRoots": ["node_modules/@types"],
+    "types": []
```
Comment (jelbourn) [inline tools/dashboard/src/tsconfig.json]: And `noImplicitAny`

File: tools/dashboard/src/app/app.module.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule {}
```
Comment (jelbourn) [inline tools/dashboard/src/app/app.module.ts]: `DashboardModule`

File: tools/dashboard/src/app/app.component.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
```
Comment (jelbourn) [inline tools/dashboard/src/app/app.component.ts]: `DashboardApp`

File: tools/dashboard/src/app/app.component.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
```
Comment (jelbourn) [inline tools/dashboard/src/app/app.component.ts]: `dashboard-app`

**Comment by DevVersion** [comment]
@jelbourn Done. Addressed all your comments.

**Pull request merged**